src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -10766,6 +10766,145 @@ .maxstack 8
             CompileAndVerify(comp, expectedOutput: "System.Action`2[System.Int32[],System.Int32]");
         }
 
+        [ConditionalFact(typeof(CoreClrOnly))]
+        [WorkItem(63565, "https://github.com/dotnet/roslyn/issues/63565")]
+        public void SynthesizedDelegateTypes_UnscopedRefAttribute_01()
+        {
+            string source = """
+                using System;
+                using System.Diagnostics.CodeAnalysis;
+                class Program
+                {
+                    static ref int F1(ref int x, ref int y) => ref x;
+                    static ref int F2(ref int x, [UnscopedRef] ref int y) => ref x;
+                    static ref int F3([UnscopedRef] ref int x, ref int y) => ref x;
+                    static void Main()
+                    {
+                        var d1 = F1;
+                        var d2 = F2;
+                        var d3 = F3;
+                        var d4 = ([UnscopedRef] ref int x, ref int y) => ref x;
+                        var d5 = (ref int x, [UnscopedRef] ref int y) => ref x;
+                        var d6 = (ref int x, ref int y) => ref x;
+                        Report(d1);
+                        Report(d2);
+                        Report(d3);
+                        Report(d4);
+                        Report(d5);
+                        Report(d6);
+                    }
+                    static void Report(Delegate d) => Console.WriteLine(d.GetType());
+                }
+                """;
+            var verifier = CompileAndVerify(source, targetFramework: TargetFramework.Net70, verify: Verification.Skipped, expectedOutput:
+                """
+                <>F{00000015}`3[System.Int32,System.Int32,System.Int32]
+                <>f__AnonymousDelegate0
+                <>f__AnonymousDelegate1
+                <>f__AnonymousDelegate1
+                <>f__AnonymousDelegate0
+                <>F{00000015}`3[System.Int32,System.Int32,System.Int32]
+                """);
+            verifier.VerifyTypeIL("<>f__AnonymousDelegate1",
```

Comment:
>f__AnonymousDelegate1 [](http://example.com/codeflow?start=37&length=21)

Is f__AnonymousDelegate0 not interesting?

Reply:
`f__AnonymousDelegate0` is inferred from a method group without `[UnscopedRef]`.

Reply:
It looks like we are using the attribute on different parameters (first and the second)

Reply:
Sorry, I'd overlooked the differently named delegate type before `<>f__AnonymousDelegate0`. Added `<>f__AnonymousDelegate0`.